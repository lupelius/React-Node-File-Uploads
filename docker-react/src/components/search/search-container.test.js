// search-container.test.js
import React from 'react';
import TextField from '@material-ui/core/TextField';
import renderer from 'react-test-renderer';

test('Component that fires onChange event, and matches snapshot', () => {
  const handleFilter = event => {};
  const component = renderer.create(
    <div className="Search">
      <TextField style={{padding: 24}}
        id="searchInput"
        placeholder="Search for Files"   
        margin="normal"
        onChange={handleFilter}
      />
    </div>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});