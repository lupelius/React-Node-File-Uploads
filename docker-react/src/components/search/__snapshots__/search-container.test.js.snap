// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component that fires onChange event, and matches snapshot 1`] = `
<div
  className="Search"
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    style={
      Object {
        "padding": 24,
      }
    }
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="searchInput"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Search for Files"
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`Component that fires onChange event, and matches snapshot 2`] = `
<div
  className="Search"
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    style={
      Object {
        "padding": 24,
      }
    }
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="searchInput"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Search for Files"
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`Component that fires onChange event, and matches snapshot 3`] = `
<div
  className="Search"
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    style={
      Object {
        "padding": 24,
      }
    }
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="searchInput"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Search for Files"
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;
